### PR TITLE
Update github.com/docker/distribution to v2.7.1

### PR DESCRIPTION
**Pull request by dims**
We should be using specific version numbers rather than random SHA(s). the closest we have to what we use is `v2.7.1` for github.com/docker/distribution

**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Review by BenTheElder — APPROVED** [review]
/lgtm
/approve

File: vendor/github.com/docker/distribution/registry/api/errcode/handler.go [hunk 1]
```diff
@@ -36,9 +36,5 @@ func ServeJSON(w http.ResponseWriter, err error) error {
 
 	w.WriteHeader(sc)
 
-	if err := json.NewEncoder(w).Encode(err); err != nil {
-		return err
-	}
-
-	return nil
+	return json.NewEncoder(w).Encode(err)
```
Comment (BenTheElder) [inline vendor/github.com/docker/distribution/registry/api/errcode/handler.go]: this is _really_ close to what we had, nice @dims 🙃

Comment (dims) [reply vendor/github.com/docker/distribution/registry/api/errcode/handler.go]: LOL

**Pull request merged**